### DISCUSSION TITLE
read_opus time data format warning

### DISCUSSION THREAD

**Discussion by udiland** [spectrochempy/spectrochempy#779] [Q&A] [answered]
Hi,

I have opus files (~300) I want to load using the 'read_opus' function.
I get a warning that I don't understand and couldn't find a solution.

WARNING | (UserWarning) time data '25/03/2020_10:31:19.-70' does not match format '%d/%m/%Y_%H:%M:%S.%f'

It looks like the negative number at the end is what making the problem but I don't know where it came from or how to change it.

The object that returns is 'NoneType'

Thanks for your help

**Comment by fernandezc** [comment]
Hi,

Sorry for the very late reply, and thanks for reporting this.

This looks like a real bug in the OPUS reader: the acquisition timestamp is parsed with a strict datetime format, but some OPUS files appear to contain a malformed or instrument-specific fractional part such as:

`25/03/2020_10:31:19.-70`

This should not make `read_opus()` return `None`. At minimum, SpectroChemPy should ignore or sanitize this malformed fractional part and still return the dataset, possibly with a warning if the acquisition date cannot be recovered exactly.

We will convert this discussion into a bug/issue so it can be tracked properly. If you still have one small representative OPUS file showing the problem, it would be very useful for adding a regression test.

Thanks again, and sorry this remained unanswered for so long.

**Comment by fernandezc** [comment, marked as answer]
This has been solved in the last dev version and will be released soon. I am closing this discussion now as resolved.
